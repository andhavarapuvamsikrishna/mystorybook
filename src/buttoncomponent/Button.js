import React,{Component} from "react"
import { action } from "@storybook/addon-actions"
class Button extends Component
{
    constructor()
    {
        super()
        this.state={
            count:0
        }
        this.handleChange=this.handleChange.bind(this)
        this.handleChange2=this.handleChange2.bind(this)
    }

    handleChange()
    {
        this.setState(function(prevState){
            //console.log(prevState.count)
             return {count:prevState.count+1}
            
        })
    }
    handleChange2()
    {
        this.setState(function(prevState){
            //console.log(prevState.count)
             return {count:prevState.count-1}
            
        })
    }
    render()
    {
        return(
        <div style={{textAlign:"center"}}>
        <h1 data-test="counter1">{this.state.count}</h1>
        <button value="increment" data-test="buttoni" onClick={action('clicked')} onClick={this.handleChange}>increment</button>
        
        <button value="decrement" data-test="buttond" onClick={action('clicked')} onClick={this.handleChange2} >decrement</button>
        </div>
        )
    } 
}
export default Button