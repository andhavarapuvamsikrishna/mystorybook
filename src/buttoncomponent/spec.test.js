import React from "react"
import Button from "./Button.js"
import {shallow} from "enzyme"
import { findByTestAttr } from "../utils/find.js"
import renderer from "react-test-renderer"
const setUp=(props={}) =>{
    const component = shallow(<Button {...props} />)
    return component
}

describe("my button component",() =>
{
    it("matches the snapshot",() =>{
        const tree = renderer.create(<Button />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it("starts with count 0",()=>
    {
        let wrapper=setUp();
        const counter=findByTestAttr(wrapper,"counter1")
        expect(counter.text()).toEqual("0")  
        //const counter=wrapper.state().count
        //console.debug(counter)
        //expect(counter).toBe(0)
    })   
    it("should increment the count when the button is clicked",() =>{
        let wrapper = setUp()
        const buttoni=findByTestAttr(wrapper,"buttoni")
        buttoni.simulate("click")
        const counter=findByTestAttr(wrapper,"counter1")
        expect(counter.text()).toEqual("1")  
    })
    it("should decrement the count when th button is clicked",() =>{
        let wrapper=setUp()
        const buttond=findByTestAttr(wrapper,"buttond")
        buttond.simulate("click")
        const counter=findByTestAttr(wrapper,"counter1")
        expect(counter.text()).toEqual("-1")
    })
})           