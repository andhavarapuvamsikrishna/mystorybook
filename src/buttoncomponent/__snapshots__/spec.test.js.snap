// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my button component matches the snapshot 1`] = `
<div
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <h1
    data-test="counter1"
  >
    0
  </h1>
  <button
    data-test="buttoni"
    onClick={[Function]}
    value="increment"
  >
    increment
  </button>
  <button
    data-test="buttond"
    onClick={[Function]}
    value="decrement"
  >
    decrement
  </button>
</div>
`;
